// @ton/sandbox/dist/utils/message.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.internal = void 0;
const core_1 = require("@ton/core");
const ec_1 = require("./ec");
/**
 * Creates {@link Message} from params.
 */
function internal(params) {
    let ecd = undefined;
    if (params.ec !== undefined) {
        if (Array.isArray(params.ec)) {
            ecd = (0, ec_1.packEc)(params.ec);
        }
        else if (params.ec instanceof core_1.Dictionary) {
            ecd = params.ec;
        }
        else {
            ecd = (0, ec_1.packEc)(Object.entries(params.ec).map(([k, v]) => [Number(k), v]));
        }
    }
    return {
        info: {
            type: 'internal',
            dest: params.to,
            src: params.from,
            value: { coins: params.value, other: ecd },
            bounce: params.bounce ?? true,
            ihrDisabled: params.ihrDisabled ?? true,
            bounced: params.bounced ?? false,
            ihrFee: params.ihrFee ?? 0n,
            forwardFee: params.forwardFee ?? 0n,
            createdAt: params.createdAt ?? 0,
            createdLt: params.createdLt ?? 0n,
        },
        body: params.body ?? new core_1.Cell(),
        init: params.stateInit,
    };
}
exports.internal = internal;
